packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -55,21 +57,78 @@ export class GitClientService {
   async createPullRequest(
     createPullRequestArgs: CreatePullRequestArgs
   ): Promise<string> {
-    const { owner, repositoryName, pullRequestModule, gitResourceMeta } =
-      createPullRequestArgs;
+    const {
+      owner,
+      repositoryName,
+      branchName,
+      commitMessage,
+      pullRequestTitle,
+      pullRequestBody,
+      pullRequestMode,
+      gitResourceMeta,
+      files,
+    } = createPullRequestArgs;
     const amplicationIgnoreManger = await this.manageAmplicationIgnoreFile(
       owner,
       repositoryName
     );
-    const files = await prepareFilesForPullRequest(
+    // console.log("files before", files);
+    const preparedFiles = await prepareFilesForPullRequest(
       gitResourceMeta,
-      pullRequestModule,
+      files,
       amplicationIgnoreManger
     );
-    return this.provider.createPullRequest(createPullRequestArgs, files);
+
+    if (pullRequestMode === EnumPullRequestMode.Basic) {
+      console.log(pullRequestMode, "pullRequestMode");
+      return this.provider.createPullRequestFromFiles({
+        owner,
+        repositoryName,
+        branchName,
+        commitMessage,
+        pullRequestTitle,
+        pullRequestBody,
+        files: preparedFiles,
+      });
+    }
+
+    if (pullRequestMode === EnumPullRequestMode.Accumulative) {
+      await this.provider.createBranchIfNotExists({
+        owner,
+        repositoryName,
+        branchName,
+      }),
+        await this.provider.createCommit({
+          owner,
+          repositoryName,
+          commitMessage,
+          branchName,
+          files: preparedFiles,
+        });
+      const { defaultBranch } = await this.provider.getRepository({
+        owner,
+        repositoryName,
+      });
+      const { url } = await this.provider.getPullRequestForBranch({
+        owner,
+        repositoryName,
+        branchName,
+      });
+      if (!url) {
+        return this.provider.createPullRequestForBranch({
+          owner,
+          repositoryName,
+          pullRequestTitle,
+          pullRequestBody,
+          branchName,
+          defaultBranchName: defaultBranch,
+        });
+      }
+      return url;
+    }
   }
 
-  async getFile(file: File): Promise<GitFile> {
+  async getFile(file: GetFileArgs): Promise<GitFile> {
     return this.provider.getFile(file);
   }
```

Comment:
Why do we need a separated function for wrapping the provider function

Reply:
because we may need this function for other providers and the git service is the service that wraps the providers